scaleway/resource_k8s_cluster.go

```diff
@@ -462,6 +486,29 @@ func resourceScalewayK8SClusterRead(ctx context.Context, d *schema.ResourceData,
 	_ = d.Set("open_id_connect_config", clusterOpenIDConnectConfigFlatten(cluster))
 	_ = d.Set("auto_upgrade", clusterAutoUpgradeFlatten(cluster))
 
+	// private_network
+	if cluster.PrivateNetworkID != nil {
+		// Temporary: the VPC API is zoned unlike the K8S API which is regional,
+		// we can't infer the zone from the region, so we check every zone until we find the PN.
+		vpcAPI := vpc.NewAPI(meta.(*Meta).scwClient)
+		i := 1
+		found := false
+		for ; i < 4; i++ {
+			_, err := vpcAPI.GetPrivateNetwork(&vpc.GetPrivateNetworkRequest{
+				Zone:             scw.Zone(fmt.Sprintf("%s-%d", cluster.Region.String(), i)),
+				PrivateNetworkID: *cluster.PrivateNetworkID,
+			})
+			if !is404Error(err) {
+				found = true
+				break
+			}
+		}
```

Comment:
Will it work in `nl-ams` ? Requests on `nl-ams-3` or `pl-waw-3` will probably not return 404 but a 400

Reply:
An hint `cluster.Region.GetZones()`